relayer/naive-strategy.go

```diff
@@ -232,6 +238,36 @@ func (nrs *NaiveStrategy) UnrelayedAcknowledgements(src, dst *Chain) (*RelaySequ
 
 // HandleEvents defines how the relayer will handle block and transaction events as they are emitted
 func (nrs *NaiveStrategy) HandleEvents(src, dst *Chain, events map[string][]string) {
+	// check for misbehaviour
+	if hdrs, ok := events[fmt.Sprintf("%s.%s", updateCliTag, headerTag)]; ok {
+		for i, hdr := range hdrs {
+			clientIDs := events[fmt.Sprintf("%s.%s", updateCliTag, clientIDTag)]
+
+			if src.PathEnd.ClientID == clientIDs[i] && dst.PathEnd.ClientID == clientIDs[i] {
+				hdrBytes, err := hex.DecodeString(hdr)
+				if err == nil {
+					exportedHeader, err := clienttypes.UnmarshalHeader(src.Encoding.Marshaler, hdrBytes)
+					if err == nil {
+						gotHeader, ok := exportedHeader.(*tmclient.Header)
+						if ok {
+							trustedHeader, err := src.GetLatestLightHeader()
+							if err == nil && !reflect.DeepEqual(gotHeader, trustedHeader) {
+								misbehaviour := tmclient.NewMisbehaviour(clientIDs[i], gotHeader, trustedHeader)
+								msg, err := clienttypes.NewMsgSubmitMisbehaviour(clientIDs[i], misbehaviour, src.MustGetAddress())
+								if err == nil {
+									_, _, _ = src.SendMsg(msg)
+								}
+							}
+						}
+					}
+
+				} else {
+					src.Error(err)
+				}
+			}
+		}
+	}
+
```

Comment:
We should try to write integration tests for this (we will need to update docker images)

Here is a description of how it should work:

Create two chains with the same validator and same chain-ids (we might need to refactor integration test code). Create a third chain with different validator/chain-id, construct links from chainA1 -> chainB and chainA2 -> B  (client, connection, channel). This should cause a divergence in state. 

Change one of the chainA's config to begin using the opposing client and submits an update message using its local chain state. The validator is still the same so verification should pass update checks. The chain who didn't change should detect misbehaviour and freeze the client. If successful, both chains switch to the still active client (on chainA) and the other chain attempts to cause misbehaviour (redo steps above)

Both clients on chainA's should become frozen if both implementations work

Let me know if you have questions